src/library/scala/math/Ordering.scala

```diff
@@ -224,11 +224,7 @@ object Ordering extends LowPriorityOrderingImplicits {
   implicit object Unit extends UnitOrdering
 
   trait BooleanOrdering extends Ordering[Boolean] {
-    def compare(x: Boolean, y: Boolean) = (x, y) match {
-      case (false, true) => -1
-      case (true, false) => 1
-      case _ => 0
-    }
+    def compare(x: Boolean, y: Boolean) = if(x) { if (y) 0 else 1} else {if (y) -1 else 0}
```

Comment:
In big changes like this, we like to separate concerns. The PR  should be restricted to changes that are crucial, and anything non-obvious must be motivated. Let's discuss whether to split up along module boundaries or along the kind of change. I think it makes most sense to have one commit per kind of change (procedure syntax, explicit result types for implicits, avoid name clashes for inner classes). I would also like to see a bytecode diff. We have automated this: https://github.com/retronym/libscala/tree/master/git-java (git diff invokes a custom command to generate only the essential diff). Here's an asm-based javap replacement that sorts by method name: https://github.com/adriaanm/scala/commit/afd15bb291b81dff4df4c22c384a4ab2c4714fad